custom_components/tuya_local/devices/dream_maker_feel_fan.yaml

```diff
@@ -0,0 +1,137 @@
+name: Pedestal Fan
+products:
+  - id: lf3n6whotmyd1d1q
+    name: Dream Maker Feel Fan DM01
+primary_entity:
+  entity: fan
+  translation_only_key: fan_with_presets
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 2
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: "Straight"
+          value: normal
+        - dps_val: "Natural"
+          value: nature
+        - dps_val: "AI"
+          value: smart
+    - id: 3
+      type: integer
+      name: speed
+      range:
+        min: 0
+        max: 100
+    - id: 5
+      type: boolean
+      name: oscillate
+
+secondary_entities:
+  - entity: select
+    name: Oscillation Angle
+    icon: mdi:arrow-oscillating
+    category: config
+    dps:
+      - id: 7
+        name: option
+        type: string
+        mapping:
+          - dps_val: 30
+            value: "30°"
+          - dps_val: 60
+            value: "60°"
+          - dps_val: 90
+            value: "90°"
+          - dps_val: 120
+            value: "120°"
+          - dps_val: 140
+            value: "140°"
+  - entity: switch
+    translation_key: indicator
+    name: Beep
+    icon: mdi:bell
+    category: config
+    dps:
+      - id: 13
+        type: boolean
+        name: switch
+  - entity: lock
+    translation_key: child_lock
+    category: config
+    dps:
+      - id: 14
+        type: boolean
+        name: lock
+  - entity: switch
+    name: Lights
+    icon: mdi:lightbulb
+    category: config
+    dps:
+      - id: 15
+        type: boolean
+        name: switch
+  - entity: sensor
+    class: temperature
+    dps:
+      - id: 21
+        type: integer
+        name: sensor
+        unit: C
+        class: measurement
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    name: Fault Alarm
```

Comment:
Avoid using a name for the only (or main) problem sensor on a device. Since the value is translated to a true/false boolean in a binary sensor, also include a second attribute with the same id, and the name "fault_code" and no mapping, so the original value is available if needed.

Reply:
I'm not even sure what fault codes it can output. There was no documentation about the fans error states. I've tried physically stopping it from oscillating, force stopping the blade from spinning etc. Maybe it's worth just removing it?